pkg/sql2pgroll/create_table.go

```diff
@@ -189,6 +202,66 @@ func convertColumnDef(tableName string, col *pgq.ColumnDef) (*migrations.Column,
 	}, nil
 }
 
+func convertConstraint(c *pgq.Constraint) (*migrations.Constraint, error) {
+	var constraintType migrations.ConstraintType
+	var nullsNotDistinct bool
+
+	switch c.Contype {
+	case pgq.ConstrType_CONSTR_UNIQUE:
+		constraintType = migrations.ConstraintTypeUnique
+		nullsNotDistinct = c.NullsNotDistinct
+	default:
+		return nil, nil
+	}
+
+	columns := make([]string, len(c.Keys))
+	for i, key := range c.Keys {
+		columns[i] = key.GetString_().Sval
+	}
+
+	including := make([]string, len(c.Including))
+	for i, include := range c.Including {
+		including[i] = include.GetString_().Sval
+	}
+
+	options := make([]string, len(c.Options))
+	for i, option := range c.Options {
+		var val string
+		switch v := option.GetDefElem().Arg.GetNode().(type) {
+		case *pgq.Node_Float:
+			val = v.Float.GetFval()
+		case *pgq.Node_Integer:
+			val = fmt.Sprintf("%d", v.Integer.GetIval())
+		case *pgq.Node_String_:
+			val = v.String_.GetSval()
+		case *pgq.Node_Boolean:
+			val = v.Boolean.String()
+		default:
+			return nil, fmt.Errorf("unsupported storage parameter type: %T", v)
+		}
+		options[i] = fmt.Sprintf("%s = '%s'", option.GetDefElem().Defname, val)
+	}
```

Comment:
Storage parameters can be deparsed directly, as done elsewhere in `sql2pgroll`:

https://github.com/xataio/pgroll/blob/daba767a10c244cc2efcae982dc11c67fdfd61a2/pkg/sql2pgroll/create_index.go#L52-L60

The ability to directly deparse some AST subtrees was added to the fork of `pg_query_go`:

https://github.com/xataio/pg_query_go/pulls?q=is%3Apr+is%3Aclosed

Reply:
Yay, this is a great addition to `pg_query_go`. \o/